libnopegl/src/pipeline_compat.c

```diff
@@ -228,7 +243,10 @@ void ngli_pipeline_compat_draw_indexed(struct pipeline_compat *s, const struct b
     struct gpu_ctx *gpu_ctx = s->gpu_ctx;
     struct pipeline *pipeline = s->pipeline;
     ngli_gpu_ctx_set_pipeline(gpu_ctx, pipeline);
-    ngli_gpu_ctx_draw_indexed(gpu_ctx, indices, indices_format, nb_indices, nb_instances);
+    for (size_t i = 0; i < s->nb_vertex_buffers; i++)
+        ngli_gpu_ctx_set_vertex_buffer(gpu_ctx, (uint32_t)i, s->vertex_buffers[i]);
+    ngli_gpu_ctx_set_index_buffer(gpu_ctx, indices, indices_format);
```

Comment:
So we're going toward a stateful approach similar to opengl? Can you elaborate on the overall intent/project?

Reply:
My intent is to decorrelate the ressources management from the pipeline to allow binding a pipeline once and execute multiple time with different data in a more saner way than what we do currently (ie: not having the pipeline object encapsulating everything). The API I lean toward is the WGPU API.